students/ChiHo/session03/mailroom.py

```diff
@@ -0,0 +1,131 @@
+# Mailroom Lab
+# Student: Chi Kin Ho
+# Date: Thursday, January 28, 2016
+
+
+def display_menu():
+
+    """
+        Display the menu of 3 actions: (1) Send a Thank You, (2) Create a Report, or (3) Quit.  If the user
+        enters an action other than 1, 2, or 3, the program will prompt him/her to enter an action again
+        until the correct action is entered.
+
+        :return: 1, 2 or 3 which corresponds to each of the above three actions.
+    """
+
+    # Repeatedly display a menu of 3 actions until the user enters the correct action.
+    while True:
+        print('\nMenu of 3 Actions')
+        print('\t1. Send a Thank You')
+        print('\t2. Create a Report')
+        print('\t3. Quit')
+        action = input('Enter an action: ')
+
+        if action == '1' or action == '2' or action == '3':
+            return int(action) # action is correct
+        else:
+            print('Invalid input.  Please enter either 1, 2 or 3')
+
+
+def send_a_thank_you(donors_dict):
+
+    while True:
+        name = input("Enter the donor's name: " )
+        if name == 'list':
+            # Display the list of donors' names.
+            for donor_name in sorted(donors_dict):
+                print(donor_name)
+        else:
+            # Prompt the user for a donation amount.
+            donation_amount = input("Enter a donation amount: $")
+
+            while not donation_amount.isdecimal():
+                # Re-prompt the user for a donation amount.
+                donation_amount = input("Enter a donation amount: $")
+
+            # Round the donation amount to 2 decimal places.
+            donation_amount = round(float(donation_amount), 2)
+
+            # Round the donation ammount to 2 decimal places.
+            donation_amount = round(donation_amount, 2)
+            if name in donors_dict: # The donor's name is already in the dictionary.
+                # Update the donor's record.
+                history_of_donations = donors_dict[name]
+                history_of_donations[0] += donation_amount # total donated
+                history_of_donations[1] += 1 # number of donations
+                history_of_donations[2] = round(history_of_donations[0] / history_of_donations[1], 2) # average
+            else: # New donor
+                # Add the new donor's name and the donation amount to the dictionary.
+                donors_dict[name] = [donation_amount, 1, donation_amount]
+            # Compose an e-mail thanking the donor for his/her generous donation, and print
+            # it on the terminal.
+            print('Thank you, {}, for your generous donation!'.format(name))
+
+            # The sending of a thank-you is completed!
+            return donors_dict
+
+
+def create_a_report(donors_dict):
```

Comment:
I get an infinite loop when I try to generate the report. Is this working for you?